src/invidious/videos.cr

```diff
@@ -909,6 +909,10 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
         "reason" => JSON::Any.new(reason),
       }
     end
+  elsif video_id != player_response.dig("videoDetails", "videoId")
+    # YouTube may return a different video player response than expected.
+    # See: https://github.com/TeamNewPipe/NewPipe/issues/8713
+    raise VideoNotAvailableException.new("Different video returned by YouTube compared to the one requested. Can't playback the video.")
```

Comment:
I'd reword the first sentence as follows:
"The video returned by Youtube isn't the requested one"

We can also remove the second sentence, as it will be obvious that the video can't be played (the user is on the error page).

Reply:
Reworded